code/modules/reagents/chemistry/machinery/reagentgrinder.dm

```diff
@@ -331,7 +363,7 @@
 						var/space = beaker.reagents.maximum_volume - beaker.reagents.total_volume
 						var/amount = get_juice_amount(O)
 
-						beaker.reagents.add_reagent(r_id, min(amount, space))
+						beaker.reagents.add_reagent(r_id, min(amount*efficiency, space))
```

Comment:
this can have some balance implications, giving you more reagents the better the efficiency, best to update the CL with stuff like this